lib/salus/scanners/report_node_modules.rb

```diff
@@ -58,7 +62,72 @@ def self.supported_languages
 
     private
 
+    def yarn_version
+      shell_return = run_shell(YARN_VERSION_COMMAND)
+      # stdout looks like "1.22.0\n"
+      shell_return.stdout&.strip
+    end
+
     def record_dependencies_from_yarn_lock
+      if Gem::Version.new(yarn_version) >= Gem::Version.new(BREAKING_VERSION)
+        record_dependencies_from_latest_yarn_lock
+      else
+        record_dependencies_from_legacy_yarn_lock
+      end
+    end
+
+    def record_dependencies_from_latest_yarn_lock
+      packages_shell_return = run_shell(
+        YARN_LIST_PACKAGES_COMMAND, chdir: @repository.path_to_repo
+      )
+      unless packages_shell_return.success?
+        raise ParseError, "Failed to parse yarn.lock file: #{packages_shell_return.stderr}"
```

Comment:
Is it possible for the command to return errors unrelated to parsing yarn.lock?
If so, maybe update the error message?

Reply:
Could be haven't seen it yet. Yes let update the error message.